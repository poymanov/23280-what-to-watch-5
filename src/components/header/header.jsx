import React, {PureComponent} from "react";
import {connect} from "react-redux";
import UserTypes from "../../types/user";
import AuthUser from "../auth-user/auth-user";
import NotAuthUser from "../not-auth-user/not-auth-user";
import {Link} from "react-router-dom";
import {checkAuth} from "../../store/api-actions";
import PropTypes from "prop-types";

class Header extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.checkAuth();
  }

  render() {
    const {currentUser, children} = this.props;

    const userBlock = currentUser ? <AuthUser currentUser={currentUser}/> : <NotAuthUser/>;

    return (
      <header className="page-header user-page__head">
        <div className="logo">
          <Link to="/" className="logo__link">
            <span className="logo__letter logo__letter--1">W</span>
            <span className="logo__letter logo__letter--2">T</span>
            <span className="logo__letter logo__letter--3">W</span>
          </Link>
        </div>
        {children}
        {userBlock}
      </header>
    );
  }
}

Header.propTypes = {
  currentUser: UserTypes.currentUser,
  checkAuth: PropTypes.func.isRequired,
  children: PropTypes.node,
};

const mapStateToProps = (state) => ({
  currentUser: state.USER.currentUser,
});

const mapDispatchToProps = (dispatch) => ({
  checkAuth() {
    dispatch(checkAuth());
  },
});

export {Header};
export default connect(mapStateToProps, mapDispatchToProps)(Header);
