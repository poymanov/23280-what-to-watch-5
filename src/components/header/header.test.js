import React from "react";
import renderer from "react-test-renderer";
import {Header} from "./header";
import {Router as BrowserRouter} from "react-router-dom";
import browserHistory from "../../etc/browser-history";

const noop = () => {};

const currentUser = {
  id: 1,
  name: `test`,
  email: `test`,
  avatarUrl: `test`
};

describe(`Should Header render correctly`, () => {
  it(`Not auth`, () => {
    const tree = renderer
      .create(
          <BrowserRouter history={browserHistory}>
            <Header checkAuth={noop} currentUser={null}/>
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Auth`, () => {
    const tree = renderer
      .create(
          <BrowserRouter history={browserHistory}>
            <Header checkAuth={noop} currentUser={currentUser}/>
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With content`, () => {
    const tree = renderer
      .create(
          <BrowserRouter history={browserHistory}>
            <Header checkAuth={noop} currentUser={null}><span>test</span></Header>
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});


