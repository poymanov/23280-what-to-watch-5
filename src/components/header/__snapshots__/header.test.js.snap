// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Header render correctly Auth 1`] = `
<header
  className="page-header user-page__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
    >
      <a
        href="/my-list"
        onClick={[Function]}
      >
        <img
          alt="test"
          height="63"
          src="test"
          width="63"
        />
      </a>
    </div>
  </div>
</header>
`;

exports[`Should Header render correctly Not auth 1`] = `
<header
  className="page-header user-page__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <a
      className="user-block__link"
      href="/login"
      onClick={[Function]}
    >
      Sign in
    </a>
  </div>
</header>
`;

exports[`Should Header render correctly With content 1`] = `
<header
  className="page-header user-page__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <span>
    test
  </span>
  <div
    className="user-block"
  >
    <a
      className="user-block__link"
      href="/login"
      onClick={[Function]}
    >
      Sign in
    </a>
  </div>
</header>
`;
